From b9760919eb3f805dd3d2af5692b49cba9c468595 Mon Sep 17 00:00:00 2001
From: "Hongzhi.Song" <hongzhi.song@windriver.com>
Date: Mon, 26 Nov 2018 02:39:59 -0500
Subject: [PATCH] kdat: Don't open /run/criu.kdat if doesn't exists

When CRIU is called for a first time and the /run/criu.kdat file does
not exists, the following warning is shown:
	Warn  (criu/kerndat.c:847): Can't load /run/criu.kdat

This patch is replacing this warning with a more appropriate debug
message.
	File /run/criu.kdat does not exist

Signed-off-by: Radostin Stoyanov <rstoyanov1@gmail.com>
Signed-off-by: Andrei Vagin <avagin@virtuozzo.com>

Upstream-Status: Backport [https://github.com/checkpoint-restore/criu/commit/868de91a099c00f099d8a41557d660d173f72521]

Signed-off-by: Hongzhi.Song <hongzhi.song@windriver.com>
---
 criu/kerndat.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/criu/kerndat.c b/criu/kerndat.c
index 8cb5690..9fb475c 100644
--- a/criu/kerndat.c
+++ b/criu/kerndat.c
@@ -789,7 +789,10 @@ static int kerndat_try_load_cache(void)
 
 	fd = open(KERNDAT_CACHE_FILE, O_RDONLY);
 	if (fd < 0) {
-		pr_warn("Can't load %s\n", KERNDAT_CACHE_FILE);
+		if(ENOENT == errno)
+			pr_debug("File %s does not exist\n", KERNDAT_CACHE_FILE);
+		else
+			pr_warn("Can't load %s\n", KERNDAT_CACHE_FILE);
 		return 1;
 	}
 
-- 
2.8.1

